docs/upgrade-to-v4.md

```diff
@@ -210,6 +210,35 @@ Since the HTTP status code and response headers are now available through the Gr
 
 The `IGraphServiceClient` interface is not really an interface because it continues to change with metadata changes. This makes it not ideal to mock or inherit. The interface has therefore been removed and no longer exists.
 
+To alleviate challenges brought about by using mocking frameworks(such Moq), the properties/methods of the `GraphServiceClient` have been made virtual. 
```

Comment:
if we get around to removing the interfaces for the request builders, we might need to to that for request builders properties as well.